pkg/module_manager/module_manager.go

```diff
@@ -1141,3 +1225,41 @@ func loadStaticValues(moduleName, modulePath string) (utils.Values, error) {
 
 	return utils.LoadValuesFileFromDir(modulePath)
 }
+
+// queueHasPendingModuleRunTaskWithStartup returns true if queue has pending tasks
+// with the type "ModuleRun" related to the module "moduleName" and DoModuleStartup is set to true.
+func queueHasPendingModuleRunTaskWithStartup(q *queue.TaskQueue, moduleName string) bool {
+	if q == nil {
+		return false
+	}
+	modules := modulesWithPendingModuleRun(q)
+	meta, has := modules[moduleName]
+	return has && meta.doStartup
+}
+
+// modulesWithPendingModuleRun returns names of all modules in pending
+// ModuleRun tasks. First task in queue considered not pending and is ignored.
+func modulesWithPendingModuleRun(q *queue.TaskQueue) map[string]struct{ doStartup bool } {
+	if q == nil {
+		return nil
+	}
+
+	modules := make(map[string]struct{ doStartup bool })
```

Comment:
This map looks so complicated. Why we couldn't use `map[string]bool` ?

Reply:
I also find this piece of code debatable, for several realons:
* it's almost a complete copy of https://github.com/flant/addon-operator/blob/a9a25d81587258f0196cf315cf0ecd064055ec73/pkg/addon-operator/queue.go#L11-L45, I decided to copy it just in favor of not cluttering the PR with a lot of refactoring (it may seem it makes sense to move some queue-related functions to a dedicated util file, may it not?)
* as of map[string]struct - queues tasks have a lot different meta, at this point we need only DoStartup flag, while next time we could need something else, this is why I decided to have a go at using an anonymous struct